kong/client.go

```diff
@@ -111,8 +121,39 @@ type Status struct {
 	ConfigurationHash string `json:"configuration_hash,omitempty" yaml:"configuration_hash,omitempty"`
 }
 
-// NewClient returns a Client which talks to Admin API of Kong
-func NewClient(baseURL *string, client *http.Client) (*Client, error) {
+type StatusMessage struct {
+	Message string `json:"message"`
+}
+
+type RequestOptions struct {
+	BaseURL   *string
+	StatusURL *string
+}
+
+func parseStatusListen(listen string) string {
+	re := regexp.MustCompile(`^([\w\.:]+)\s*(.*)?`)
+	matches := re.FindStringSubmatch(listen)
+
+	if len(matches) == 0 {
+		return ""
+	}
+
+	address := matches[1]
+	extraParams := matches[2]
```

Comment:
What if `matches` is 1 or 2 entries long? This will blow up then. Can we check for that as well here?